### DISCUSSION TITLE
responseProps

### DISCUSSION THREAD

**Discussion by sauyeung** [RobinTail/express-zod-api#1049] [General]
I am testing an endpoint that uses the express-rate-limit middleware, and pass `responseProps` to `testEndpoint` as follows:
>          responseProps: {
>             ip: '192.168.1.1',
>             setHeader: (name: string, value: string) => {
>               console.log(name + ': ' + value);
>             }
>           }
The `responseMock` object does have these attributes added when testing cases of this endpoint that are expected to succeed; but not added when they are expected to fail; and got _response.setHeader is not a function_ error. Is this an issue or expected? If it's expected, are there any ways to make it work?

**Comment by sauyeung** [comment]
Could we add `setHeader` and `send` to `makeResponseMock` under https://github.com/RobinTail/express-zod-api/blob/master/src/mock.ts?

**Comment by RobinTail** [comment]
> Could we add setHeader and send to makeResponseMock

Yep, we can.

**Comment by RobinTail** [comment]
https://github.com/RobinTail/express-zod-api/pull/1098 , @sauyeung

**Comment by RobinTail** [comment]
🚀 No manual mocking of request/response needed in the recently released v20, @sauyeung
